optuna/study.py

```diff
@@ -848,6 +849,55 @@ def objective(trial):
 
         self._storage.create_new_trial(self._study_id, template_trial=trial)
 
+    @experimental("2.5.0")
+    def add_trials(self, trials: Iterable[FrozenTrial]) -> "Study":
+        """Add trials to study.
+
+        The trials are validated before being added.
+
+        Example:
+
+            .. testcode::
+
+                import optuna
+                from optuna.distributions import UniformDistribution
+
+
+                def objective(trial):
+                    x = trial.suggest_uniform("x", 0, 10)
+                    return x ** 2
+
+
+                study = optuna.create_study()
+                study.optimize(objective, n_trials=3)
+                assert len(study.trials) == 3
+
+                other_study = optuna.create_study().add_trials(study.trials)
+                assert len(other_study.trials) == len(study.trials)
+
+                other_study.optimize(objective, n_trials=2)
+                assert len(other_study.trials) == len(study.trials) + 2
+
+        .. seealso::
+
+            See :func:`~optuna.study.Study.add_trial` for addition of each trial.
+
+        Args:
+            trials: Trials to add.
+
+        Returns:
+            Return self.
```

Comment:
@not522 @sile I prefer that `add_trials()` method returns nothing because of following reasons.

1. This method provides in-place operation.
1. `add_trial()` method does not return anything.

What do you think?

Reply:
I'm neutral about this point.
However, indeed, I also felt the return type of `add_trials()` was inconsistent with the one of `add_trial()` (so if we allow `add_trials()` returning `self`, it might be better to modify `add_trial()` as well).

Reply:
Thank you for your suggestion. To keep consistency with other methods, I changed the return value as None. PTAL.